### PR TITLE
Avoid creating SafeAggregateSink wrapper around empty list

### PR DESCRIPTION
**Pull request by sungam3r**

### CODE REVIEW

File: src/Serilog/Core/Sinks/SafeAggregateSink.cs [hunk 1]
```diff
@@ -18,6 +18,8 @@ class SafeAggregateSink : ILogEventSink
 {
     readonly ILogEventSink[] _sinks;
 
+    internal static SafeAggregateSink Empty = new(new ILogEventSink[0]);
```
Comment (nblumhardt) [inline src/Serilog/Core/Sinks/SafeAggregateSink.cs:21]: This appears to be unused?

File: src/Serilog/LoggerConfiguration.cs [hunk 2]
```diff
@@ -128,11 +128,17 @@ public Logger CreateLogger()
...
+            sink = new SafeAggregateSink(_logEventSinks);
 
         var auditing = _auditSinks.Any();
         if (auditing)
-            sink = new AggregateSink(new[] { sink }.Concat(_auditSinks));
+        {
+            sink = new AggregateSink(sink == null ? _auditSinks : new[] { sink }.Concat(_auditSinks));
+        }
+
+        sink ??= SafeAggregateSink.Empty;
```
Comment (SimonCropp) [inline src/Serilog/LoggerConfiguration.cs:141]: @nblumhardt isnt it used here?

File: src/Serilog/Core/Sinks/SafeAggregateSink.cs (hunk 1 above)
Comment (sungam3r) [reply src/Serilog/Core/Sinks/SafeAggregateSink.cs:21]: `sink ??= SafeAggregateSink.Empty;` in another file.

File: src/Serilog/LoggerConfiguration.cs (hunk 2 above)
Comment (nblumhardt) [reply src/Serilog/LoggerConfiguration.cs:141]: Ah, my bad :-) sorry!

**Comment by nblumhardt** [comment]
In the majority of applications, where there are a nonzero number of sinks, wouldn't this just lead to an additional/wasted/long-lived instance of `SafeAggregateSink` and an empty array?

**Comment by sungam3r** [comment]
I removed field.

**Comment by nblumhardt** [comment]
LGTM! 👍

**Pull request merged**